### PR TITLE
000A-Front-end-Setup (Authorisation,Authentication,Accessibility)

### PR DESCRIPTION
**Pull request by Adnanhoq**

### CODE REVIEW

**Review by sdagn1 — APPROVED** [review]

File: src/controllers/AuthController.ts
Comment (jemorak) [inline src/controllers/AuthController.ts:1]: Mohammed you need to change the redirect name to  something else. We are not doing orders. Maybe make a random home page.

File: views/loginForm.html [hunk 1]
```diff
@@ -0,0 +1,29 @@
+<!DOCTYPE html>
+<html lang="en">
+    <head>
+        <title>Login</title>
+    </head>
+    <body>
+        <h1>Login</h1>
+
+        {% if errormessage %}
+            <h3>{{ errormessage }}</h3>
```
Comment (JennicaMistry) [inline views/loginForm.html:10]: Have you ran any accessibility testing on this?

File: src/controllers/AuthController.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+import { getToken } from "../services/AuthService"
+
+export const getLoginForm = async (req: express.Request, res: express.Response): Promise<void> => {
+    res.render('LoginForm.html');
+}
+
+export const postLoginForm = async (req: express.Request, res: express.Response): Promise<void> => {
+    try {
+        req.session.token = await getToken(req.body, req.session.token);
+        res.redirect('/orders');
```
Comment (jemorak) [inline src/controllers/AuthController.ts:11]: We are not doing orders

**Review by jemorak — CHANGES_REQUESTED** [review]

File: src/controllers/AuthController.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import express from "express";
+import { getToken } from "../services/AuthService"
+
+export const getLoginForm = async (req: express.Request, res: express.Response): Promise<void> => {
+    res.render('LoginForm.html');
+}
+
+export const postLoginForm = async (req: express.Request, res: express.Response): Promise<void> => {
+    try {
+        req.session.token = await getToken(req.body, req.session.token);
```
Comment (JennicaMistry) [inline src/controllers/AuthController.ts:10]: Have you validated the data?

**Review by JennicaMistry — CHANGES_REQUESTED** [review]
Accessibility & Validation

File: src/controllers/AuthController.ts (hunk 3 above)
Comment (Adnanhoq) [reply src/controllers/AuthController.ts:10]: It is being validated in the backend - Sam

File: views/loginForm.html (hunk 1 above)
Comment (Adnanhoq) [reply views/loginForm.html:10]: I will add that now

File: src/controllers/AuthController.ts (hunk 3 above)
Comment (JennicaMistry) [reply src/controllers/AuthController.ts:10]: I can see the Validator

**Review by Leebee1392 — APPROVED** [review]

**Review by sdagn1 — APPROVED** [review]

**Pull request merged**